eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Generator/Trace/Chain.hs

```diff
@@ -102,9 +103,10 @@ instance
   , Environment (EraRule "TICK" era) ~ ()
   , State (EraRule "TICK" era) ~ NewEpochState era
   , Signal (EraRule "TICK" era) ~ SlotNo
-  , QC.HasTrace (EraRule "LEDGERS" era) (GenEnv era)
+  , QC.HasTrace (EraRule "LEDGERS" era) (GenEnv c era)
+  , c ~ MockCrypto
   ) =>
-  HasTrace (CHAIN era) (GenEnv era)
+  HasTrace (CHAIN era) (GenEnv c era)
```

Comment:
Why not: 
```suggestion
  , QC.HasTrace (EraRule "LEDGERS" era) (GenEnv MockCrypto era)
  ) => 
  HasTrace (CHAIN era) (GenEnv MockCrypto era)
```
? Any reason in particular?

Reply:
Nice catch! Thanks! I have updated the changes.